### PR TITLE
Feature/farm 479 crop repository

### PR DESCRIPTION
**Pull request by LeeMatthewHiggins**
[FARM-479 - Crop firebase repo]

#### 📲 What

Added firebase repo, crops now come from CMS and not Mock generator.

#### 🤔 Why
		
Needed to complete the CMS driven functionality. 
		
#### 🛠 How
		
Added the repo, hooked up to firebase system.

#### 👀 See
		
Screenshots / external resources
		 
#### 🕵️‍♂️ How to test

Crops and scoring will non longer match, so all crops will have 0% recommendation until the scoring firebase repo is completed.

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/model/repositories/crop/implementation/transformers/FirebaseCropStageTransformer.dart [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        this._metaTransformer = metaTransformer;
+
+  @override
+  ArticleEntity transform({DocumentSnapshot from}) {
+    final meta = _metaTransformer.transform(from: from);
+    final id = castOrNull<String>(from.data[_Fields.id]);
+    final content = castOrNull<String>(from.data[_Fields.content]);
+    final status = castOrNull<String>(from.data[_Fields.status]);
+    final summary = castOrNull<String>(from.data[_Fields.summary]);
+    final title = castOrNull<String>(from.data[_Fields.name]) ??
```
Comment (albertmiro) [inline lib/model/repositories/crop/implementation/transformers/FirebaseCropStageTransformer.dart]: Is ok to get the `_Fields.name` or we should be getting the `_Fields.title` ?

File: lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart [hunk 2]
```diff
@@ -7,74 +7,104 @@ import 'package:farmsmart_flutter/model/model/ImageURLProvider.dart';
...
+        _Strings.pathDivider +
+        imageSize[ImageEntityFields.path] +
+        _Strings.pathDivider +
+        imageFileNamePath;
+    final width = imageSize[ImageEntityFields.width];
+    final height = imageSize[ImageEntityFields.height];
+    return FlamelinkImageEntity(cms, width, height, path, []);
+  }).toList();
+  final alternateSizes = List<ImageEntity>.from(alternateSizesObjs);
+  return FlamelinkImageEntity(cms, 0, 0, imageFileNamePath, alternateSizes);
```
Comment (albertmiro) [inline lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart]: What these 0's mean? Can we move them to a const?

File: lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart [hunk 3]
```diff
@@ -7,74 +7,104 @@ import 'package:farmsmart_flutter/model/model/ImageURLProvider.dart';
...
+
+      for (var image in alternateImages) {
+        if ((image.width > targetWidth)) {
+          return _cms
+              .images(path: image.path)
+              .getDownloadURL()
+              .then((value) => value.toString());
+        }
+      }
+    }*/
```
Comment (albertmiro) [inline lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart:84]: Let's remove this commented code if we are not going to use it

File: test/data/bloc/recommendations/RecommendationsEngine_test.dart [hunk 4]
```diff
@@ -9,32 +8,94 @@ final testWeights = {"Tomatoes":{"Location":0.2,"Land Size":0.16, "Soil Type":0.
...
+
+    final maxEngine = RecommendationEngine(
+        inputFactors: maxInput,
+        inputScale: inputScale,
+        weightMatrix: harryWeights);
+        
+    final chillies = chilliesEngine.recommend("Chillies");
+    final cowPeas = cowpeasEngine.recommend("Cowpeas");
+    final max = maxEngine.recommend("max");
+    //TODO: add harry test data........
```
Comment (albertmiro) [inline test/data/bloc/recommendations/RecommendationsEngine_test.dart:98]: This should be done in another PR?

File: lib/model/repositories/crop/implementation/transformers/FirebaseCropStageTransformer.dart (hunk 1 above)
Comment (LeeMatthewHiggins) [reply lib/model/repositories/crop/implementation/transformers/FirebaseCropStageTransformer.dart]: should be in client terms I think "title"

File: lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart (hunk 3 above)
Comment (LeeMatthewHiggins) [reply lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart:84]: this is temp pending a server side fix / investigation. We want the code, but ATM the server returns no image for some alternate sizes.

File: test/data/bloc/recommendations/RecommendationsEngine_test.dart (hunk 4 above)
Comment (LeeMatthewHiggins) [reply test/data/bloc/recommendations/RecommendationsEngine_test.dart:98]: there's another PR coming that sorts this.

File: lib/model/repositories/crop/implementation/transformers/FirebaseCropStageTransformer.dart (hunk 1 above)
Comment (LeeMatthewHiggins) [reply lib/model/repositories/crop/implementation/transformers/FirebaseCropStageTransformer.dart]: I changed the var to name in the end, as its the stage name used for the title in this case.

File: lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart (hunk 2 above)
Comment (LeeMatthewHiggins) [reply lib/model/repositories/image/implementation/ImageRepositoryFlamelink.dart]: added unkownSize

**Review by albertmiro — APPROVED** [review]
LGTM!
